### PR TITLE
Scoring Mode changes

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import edu.greenblitz.robotName.commands.swerve.MoveToAmp;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+
+public class GoToAndScoreToAmp extends ParallelCommandGroup {
+
+    public GoToAndScoreToAmp() {
+        super(
+                new ScoreToAmp(),
+                new MoveToAmp()
```
Comment (yonatan-orca) [inline src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java:11]: Parallel of scoreToAmp and moveToAmp? Does score to amp wait for the right chassis pose? Or will this not work?

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 2]
```diff
@@ -81,13 +78,9 @@ public SmartJoystick getFourthJoystick() {
...
 		secondJoystick.START.whileTrue(new PanicMode());
 		secondJoystick.BACK.whileTrue(new ToggleScoringMode());
-		secondJoystick.A.whileTrue(new ScoreToAmp());
-		secondJoystick.B.whileTrue(new MoveElbowAndWristToSafe());
-		secondJoystick.X.whileTrue(new MoveShooterToAngle(PivotConstants.PresetPositions.PICK_UP.ANGLE));
-		secondJoystick.Y.whileTrue(new MoveShooterToAngle(PivotConstants.PresetPositions.TRANSFER.ANGLE));
 		secondJoystick.POV_DOWN.whileTrue(new CollectNoteToScoringMode());
 
-		mainJoystick.A.whileTrue(new GoToShootingState(ShootingPositionConstants.OPTIMAL_SHOOTING_ZONE));
+		mainJoystick.A.whileTrue(new ScoreToScoringMode());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:83]: why in init buttons?

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java (hunk 1 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java]: Again this is parallel so the note can be scored before the chassis is in scoring position

File: src/main/java/edu/greenblitz/robotName/OI.java (hunk 2 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/OI.java:83]: score to scoring mode replaces the GoToShootingState

File: src/main/java/edu/greenblitz/robotName/commands/ScoreToScoringMode.java [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+import edu.greenblitz.robotName.commands.arm.GoToAndScoreToAmp;
 import edu.greenblitz.robotName.commands.shooter.shootingState.GoToAndShootToSpeaker;
 import edu.wpi.first.wpilibj2.command.ConditionalCommand;
 
 public class ScoreToScoringMode extends ConditionalCommand {
 
     public ScoreToScoringMode(){
         super(
-                new ScoreToAmp(),
+                new GoToAndScoreToAmp(),
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/ScoreToScoringMode.java:12]: rename GoTo

File: src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import edu.greenblitz.robotName.commands.arm.roller.runByPower.RunRollerClockwiseUntilObjectIsOut;
+import edu.greenblitz.robotName.commands.swerve.MoveToAmp;
+import edu.greenblitz.robotName.subsystems.arm.elbow.ElbowConstants;
+import edu.greenblitz.robotName.subsystems.arm.wrist.WristConstants;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class GoToAndScoreToAmp extends SequentialCommandGroup {
+
+    public GoToAndScoreToAmp() {
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/GoToAndScoreToAmp.java:12]: rename

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Comment by yonnbro** [comment]
What’s up with this PR?

File: src/main/java/edu/greenblitz/robotName/commands/arm/MoveToAndScoreToAmp.java
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/arm/MoveToAndScoreToAmp.java:1]: you need to do another move to pose to pre score amp pose or something

**Review by aaa-png — CHANGES_REQUESTED** [review]

**Pull request closed without merging**